testonly/hammer/hammer.go

```diff
@@ -284,6 +285,75 @@ func newWorker(cfg *MapConfig, prng *rand.Rand) *mapWorker {
 	}
 }
 
+// readWorker performs read-only operations on a fixed map.
+type readWorker struct {
+	*mapWorker
+
+	validReadOps   *validReadOps
+	invalidReadOps *invalidReadOps
+}
+
+func newReadWorker(s *hammerState, idx int) *readWorker {
+	return &readWorker{
+		mapWorker: newWorker(s.cfg, rand.New(rand.NewSource(int64(idx)))),
+
+		validReadOps:   s.validReadOps,
+		invalidReadOps: s.invalidReadOps,
+	}
+}
+
+func (w *readWorker) loopUntilDone(ctx context.Context, done <-chan struct{}) error {
+	for {
+		select {
+		case <-done:
+			return nil
+		default:
+		}
+		if err := w.validReadOps.getLeavesRev(ctx, w.prng); err != nil {
+			if _, ok := err.(errSkip); ok {
```

Comment:
nit to the previous code: I'd define `var errSkip = errors.New("skip something something")` and just compare it for equality.

Reply:
errSkip is used in quite a few places and is very orthogonal to this PR. Will leave for now. I may well come through and change up `errSkip` and `testonly.ErrInvariant` which have limited/zero uses. But this is not the time nor place.